### PR TITLE
GSettingsHandler bridge can still cause core dumps for undefined keys.

### PR DESCRIPTION
**Pull request by klown**
@sgithens @javihernandez The problem is in the gpii.gsettings.setSingleKey() and gpii.gsettings.getSingleKey() functions. This pull fixes that.

### CODE REVIEW

File: gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js [hunk 1]
```diff
@@ -209,6 +211,10 @@ jqUnit.test("Setting single keys via gsettings handler", function () {
     gpii.gsettings.setSingleKey("net.gpii.testing.gsettings.single-set", "string-setting", "Absolutely awesome");
     jqUnit.assertEquals("Checking 'string' key", "Absolutely awesome",
         gpii.gsettings.getSingleKey("net.gpii.testing.gsettings.single-set", "string-setting"));
+    // set and check non-existent key
+    gpii.gsettings.setSingleKey("net.gpii.testing.gsettings.single-set", "unicorn", "Mythical");
+    jqUnit.assertEquals("Checking non-existaent 'string' key", undefined,
```
Comment (javihernandez) [inline gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js]: typo, "non-existaent"

File: gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js [hunk 2]
```diff
@@ -193,6 +193,8 @@ jqUnit.test("Getting single keys via gsettings handler", function () {
         gpii.gsettings.getSingleKey("net.gpii.testing.gsettings.single-get", "boolean-setting"));
     jqUnit.assertEquals("Checking 'string' key", "abcdefg",
         gpii.gsettings.getSingleKey("net.gpii.testing.gsettings.single-get", "string-setting"));
+    jqUnit.assertEquals("Checking non-existent' key", undefined,
```
Comment (javihernandez) [inline gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js]: there's a poor and orphan single quotation mark right after non-existent

File: gpii/node_modules/gsettingsBridge/gsettings_bridge.js [hunk 3]
```diff
@@ -41,11 +41,20 @@ fluid.defaults("gpii.gsettings.getSingleKey", {
 });
 
 gpii.gsettings.getSingleKey = function (schemaId, key) {
-    return nodeGSettings.get_gsetting(schemaId, key);
+    var keys = nodeGSettings.get_gsetting_keys(schemaId);
+    if (keys.indexOf(key) !== -1) {
+        return nodeGSettings.get_gsetting(schemaId, key);
+    }
+    else {
```
Comment (javihernandez) [inline gpii/node_modules/gsettingsBridge/gsettings_bridge.js]: pedantic mode on - can you use the form "} else {" ?

File: gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js (hunk 2 above)
Comment (klown) [inline gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js]: Good catch @javihernandez.  What's actually missing here is the type of the key.  It should be 'string' in this case. Fixed.

File: gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js (hunk 1 above)
Comment (klown) [inline gpii/node_modules/gsettingsBridge/tests/gsettingsTests.js]: Why didn't `grunt lint` find that????  Perhaps we need to add a spell checker grunt task. ;-)

File: gpii/node_modules/gsettingsBridge/gsettings_bridge.js (hunk 3 above)
Comment (klown) [inline gpii/node_modules/gsettingsBridge/gsettings_bridge.js]: Agreed -- you  are pedantic :-)  Changed.

**Comment by javihernandez** [comment]
@GPII/ci-and-infrastructure hey guys, what's going on in here with CI?

**Comment by klown** [comment]
Noting that it *did* pass [the last time CI ran correctly](https://github.com/GPII/linux/pull/90#issuecomment-276694642).  What changed since then?

**Comment by gtirloni** [comment]
ok to test

**Comment by gtirloni** [comment]
ok to test

**Comment by klown** [comment]
@javihernandez wrote:

> Thanks for catching this, I'll be happy to merge it after your update.

Done.  Actually I did it [a while ago](https://github.com/GPII/linux/pull/90/commits/5386407f0838605fe1294262685754755814d79a), but forgot to tell you.  And, the CI job passes.

Let me know if you see anything else you want changed.

**Pull request merged**